codec-http2/src/test/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributorDependencyTreeTest.java

```diff
@@ -709,10 +714,31 @@ public void closeWithNoPrioritizableDependentsShouldRestructureTree() throws Exc
         assertEquals(1, distributor.numChildren(connection.connectionStream().id()));
 
         // Level 1
-        assertTrue(distributor.isChild(streamE.id(), connection.connectionStream().id(), DEFAULT_PRIORITY_WEIGHT));
+        short halfWeight = DEFAULT_PRIORITY_WEIGHT / 2;
+        assertTrue(distributor.isChild(streamE.id(), connection.connectionStream().id(), halfWeight));
         assertEquals(0, distributor.numChildren(streamE.id()));
     }
 
+    @Test
+    public void closeStreamWithChildrenShouldRedistributeWeightToChildren() throws Exception {
+        Http2Stream streamA = connection.local().createStream(1, false);
+        Http2Stream streamB = connection.local().createStream(3, false);
+        Http2Stream streamC = connection.local().createStream(5, false);
+        Http2Stream streamD = connection.local().createStream(7, false);
+
+        setPriority(streamC.id(), streamA.id(), MAX_WEIGHT, false);
+        setPriority(streamD.id(), streamA.id(), MAX_WEIGHT, false);
```

Comment:
How about making this slightly more complex, by making each parent stream have 3 children?